dataplane-webserver/src/backups/mod.rs

```diff
@@ -0,0 +1,429 @@
+                "Destination path is not a valid S3 URI: {}",
+                destination_path
+            )))
+        }
+    } else {
+        Err(ErrorInternalServerError(
+            "CoreDB backup configuration does not contain a destination path".to_string(),
+        ))
+    }
+}
+
+/// Executes a backup using temback in the database pod.
+///
+/// This function performs the following steps:
+/// 1. Finds the primary database pod in the namespace using CloudNativePG labels
+/// 2. Checks for and installs temback binary if not present
+/// 3. Executes temback command to perform backup directly to S3
+///
+/// # Arguments
+/// * `kube_client` - Kubernetes client for pod operations
+/// * `namespace` - Namespace where the database pod is running
+/// * `bucket_name` - S3 bucket to store the backup
+/// * `bucket_path` - Base path in the S3 bucket
+///
+/// # Returns
+/// * `Ok(BackupResult)` containing Success or Failed with error message
+/// * `Err(Error)` if any step fails catastrophically
+async fn run_backup_on_instance_pod(
+    kube_client: &KubeClient,
+    namespace: &str,
+    bucket_name: &str,
+    bucket_path: &str,
+) -> Result<BackupResult, Error> {
+    // Find the primary database pod
+    let pods_api: Api<Pod> = Api::namespaced(kube_client.clone(), namespace);
+    let label_selector = format!("cnpg.io/cluster={},cnpg.io/instanceRole=primary", namespace);
+    let params = ListParams::default().labels(&label_selector);
+
+    let pods = pods_api
+        .list(&params)
+        .await
+        .map_err(|e| ErrorInternalServerError(format!("Failed to list pods: {}", e)))?;
+
+    if pods.items.is_empty() {
+        return Ok(BackupResult::Failed(
+            "No primary database pod found".to_string(),
+        ));
+    }
+
+    let pod = &pods.items[0];
+    let pod_name = pod
+        .metadata
+        .name
+        .as_ref()
+        .ok_or_else(|| ErrorInternalServerError("Pod has no name"))?;
+
+    tracing::info!(
+        namespace = %namespace,
+        pod = %pod_name,
+        "Found primary database pod for backup"
+    );
+
+    // Check if temback exists and install if needed
+    let check_cmd = vec!["test", "-f", "/var/lib/postgresql/data/temback"];
+    let attach_params = AttachParams::default()
+        .container("postgres")
+        .stderr(true)
+        .stdout(true)
+        .stdin(false);
+
+    let check_result = pods_api.exec(pod_name, check_cmd, &attach_params).await;
+    if check_result.is_err() {
+        tracing::info!(
+            pod = %pod_name,
+            "temback not found, installing..."
+        );
+
+        // Install temback
+        let install_cmd = vec![
+            "sh", 
+            "-c", 
+            "curl -L https://github.com/tembo-io/temback/releases/download/v0.1.0/temback-v0.1.0-linux-amd64.tar.gz | tar -C /var/lib/postgresql/data --strip-components=1 -zxf - temback-v0.1.0-linux-amd64/temback && chmod +x /var/lib/postgresql/data/temback"
```

Comment:
Need to update this to `0.1.1` once released. Possibly move the URL and version to a constant.